sql/lesson5/tutorial.md

```diff
@@ -0,0 +1,181 @@
+
+```SQL
+SELECT *
+  FROM hosts;
+```
+
+You will notice the `location_id` field. This field is a foreign key to the `locations` table. The value of this field for the first record, Happy Code is `2`. So let's have a look at the locations table:
+
+```SQL
+SELECT *
+  FROM locations;
+```
+
+This shows us that the record in the `locations` table that has `id` set to `2` is the location record for Brighton in the UK. That's great, we can now go from `hosts` to `locations` with two queries but can we get this data in a single query and for all records? This is where table joins come in.
+
+### Inner join
+
+#### Joining two tables
+
+If we want to select all the hosts and for each of them pull in location data, we need what we call an inner join. This is a **SELECT** query that will join the `hosts` and `locations` tables to return a single data set. Inner joins can be written either with an **INNER JOIN** or a **JOIN** clause. Both of the **SELECT** statements below result in identical SQL queries and return the same results:
+
+```SQL
+SELECT hosts.name, locations.city, locations.country
+  FROM hosts
+ INNER JOIN locations
+    ON hosts.location_id = locations.id;
+```
+
+```SQL
+SELECT hosts.name, locations.city, locations.country
+  FROM hosts
+  JOIN locations
+    ON hosts.location_id = locations.id;
+```
+
+Note that SQL is not affected by the use of white space, which is used to improve the legibility of queries. White space in SQL is commonly used to line up root keywords so that they end on the same character boundary, thus forming a river that runs down the query helping the reader to scan over the code. White space is also used with subqueries, to nest the subquery in an indented block.
+
+#### Why are these joins called 'inner' joins?
+
+The word "inner" in "inner join" refers to the fact that this type of join will only return data where records exist on both sides of the join.
+
+In other words, inner joins only add rows to the combined data set that satisfy the join condition set forth in the **ON** statement.
+
+For instance, you will notice that we don't get all locations when running the following query:
+
+```SQL
+SELECT hosts.name, locations.city, locations.country
+  FROM hosts
+  JOIN locations
+    ON hosts.location_id = locations.id;
+```
+
+Helsinki is missing because the database contains no host in Helsinki, even if there is a location record for it.
+
+#### Using WHERE with a JOIN statement
+
+We can limit the set of results using a **WHERE** clause as we did with simple **SELECT** statements:
+
+```SQL
+SELECT hosts.name, locations.city, locations.country
+  FROM hosts
+  JOIN locations
+    ON hosts.location_id = locations.id
+ WHERE locations.country = 'UK';
+```
+
+#### Joining more than two tables
+
+You can have multiple **JOIN** clauses when connecting more than two tables. So for example if we wanted to know all workshop dates with host and location, we could do:
```

Comment:
As explained above, suggest here is a good place to say that a table can have multiple foreign keys to join several tables' data to this one row.

Reply:
Good point. I'll add the following:

In the example above, **workshops** table has a foreign key (host_id) to **hosts** table, and **hosts** table has a foreign key (location_id) to **locations** table.

In some cases, you can find tables having more than one foreign key to other tables. For example, table **rsvp** has two foreign keys: one to **workshop** table (workshop_id) and one to **people** table (person_id).